docs/templates.md

```diff
@@ -0,0 +1,175 @@
+# Templates
+
+Sydent uses parametrised templates to generate the content of emails it
+sends and webpages it shows to users. Example templates can be found in
+the [res](https://github.com/matrix-org/sydent/tree/main/res) folder.
+
+## Branding
+
+Sending a value for `brand` to some API endpoints allows for different
+email and http templates to be used. These templates should be stored
+in a file structure like this:
+
+```
+ root_template_dir/
+     brand1/
+         [ brand1 template files ]
+     brand2/
+         [ brand2 template files ]
+```
+
+The config option `templates.root_directory` should be set to the path
+of `root_template_dir` and `templates.default_brand` should be set to
+the sub directory to use if no `brand` (or an invalid `brand`) is provided
+by the request.
+
+Here are the requests that can contain a value for `brand` and what
+template the brand value is used to select:
+
+Invite email templates:
+ - `POST /_matrix/identity/v2/validate/msisdn/requestToken`
+
+Verification email templates:
+ - `POST /_matrix/identity/v2/validate/email/requestToken`
+
+Verification SMS templates - BRAND CURRENTLY IGNORED:
+ - `POST /_matrix/identity/v2/validate/msisdn/requestToken`
+
+Verification response templates:
+ - `GET /_matrix/identity/v2/validate/email/submitToken`
+ - `GET /_matrix/identity/v2/validate/msisdn/submitToken`
+
+
+## Template formats
+
+### `.eml` template files
+
+Files ending in ".eml" can use python `%(variable)s` string substitution.
+
+Appending "_forurl" or "_forhtml" to any of the variable names listed below
+returns their values encoded suitably for URLs or HTML respectively.
+
+e.g. ">" in the HTML encoded version would be replaced with "&gt".
+
+### `.eml.j2` files
+
+Files ending in ".eml.j2" are Jinja templates. 
+
+Using the `urlencode` Jinja filter encodes the contents for URLs suitably.
+All variables are automatically made safe for HTML.
```

Comment:
We probably want to include that some of them should use `safe` if they're not in HTML.

Reply:
What are the advantages of marking them as safe? Is there a significant performance boost?

Reply:
It means things won't be double-escaped e.g. `>` → `&gt;` → `&amp;gt;` (renders as '&gt;').

Reply:
Where is the second layer of escaping happening?

Reply:
>Where is the second layer of escaping happening?

my mistake, have just had a very cursed history with PHP and forgot we're talking about the plaintext section of the e-mail here